libs/ui/src/components/MediaItem/MediaItem.vue

```diff
@@ -15,134 +15,128 @@
-        AudioMedia: defineAsyncComponent(() => import('./type/AudioMedia.vue')),
-        ModelMedia: defineAsyncComponent(() => import('./type/ModelMedia.vue')),
-        JsonMedia: defineAsyncComponent(() => import('./type/JsonMedia.vue')),
-        IFrameMedia: defineAsyncComponent(
-          () => import('./type/IFrameMedia.vue')
-        ),
-        ObjectMedia: defineAsyncComponent(
-          () => import('./type/ObjectMedia.vue')
-        ),
-        Media: defineAsyncComponent(() => import('./type/UnknownMedia.vue')),
-      },
-      isLewdBlurredLayer: this.isLewd,
-    }
-  },
-  computed: {
-    resolveComponent() {
-      const type = this.mimeType || this.defaultMimeType
-      return this.components[resolveMedia(type) + SUFFIX]
-    },
-    properSrc() {
-      return this.src || this.placeholder
-    },
-  },
-  watch: {
-    animationSrc() {
-      this.updateComponent()
-    },
-  },
-  mounted() {
-    this.updateComponent()
-  },
-  methods: {
-    async updateComponent() {
-      if (this.animationSrc && !this.mimeType) {
-        this.defaultMimeType = await getMimeType(this.animationSrc)
-      }
-    },
-    showContent() {
-      this.isLewdBlurredLayer = false
-    },
-  },
+let fileType = ref('image')
+let isLewdBlurredLayer = ref<boolean>(props.isLewd)
+const components = {
+  ImageMedia,
+  VideoMedia,
+  AudioMedia,
+  ModelMedia,
+  JsonMedia,
+  IFrameMedia,
+  ObjectMedia,
+  Media,
 }
-</script>
 
-<style lang="scss" scoped>
-.nsfw-blur {
-  backdrop-filter: blur(60px);
-  position: absolute;
-  top: 0;
-  height: 100%;
-  width: 100%;
-  color: #fff;
-  text-transform: capitalize;
+const resolveComponent = computed(() => {
+  const type = props.mimeType || fileType.value
+  return components[resolveMedia(type) + SUFFIX]
+})
+const properSrc = computed(() => props.src || props.placeholder)
 
-  .heading-nsfw {
-    font-weight: 700;
-  }
-  .nsfw-desc {
-    max-width: 18.75rem;
-    text-align: center;
+watch(
+  () => props.animationSrc,
+  () => updateComponent()
+)
+onMounted(() => {
+  updateComponent()
+})
+const updateComponent = async () => {
+  if (props.animationSrc && !props.mimeType) {
+    fileType.value = await getMimeType(props.animationSrc)
   }
+}
+const toggleContent = () => {
+  isLewdBlurredLayer.value = !isLewdBlurredLayer.value
+}
 
-  .nsfw-content {
-    cursor: pointer;
-    bottom: 3.125rem;
+defineExpose({ isLewdBlurredLayer })
+</script>
+
+<style lang="scss" scoped>
+@import '@/styles/abstracts/variables';
+@import '@/styles/abstracts/mixins';
+.media-object {
+  .nsfw-blur {
+    backdrop-filter: blur(60px);
     position: absolute;
+    top: 0;
+    height: 100%;
+    width: 100%;
+    color: #fff;
+    text-transform: capitalize;
+
+    .heading-nsfw {
+      font-weight: 700;
+    }
+    .nsfw-desc {
+      max-width: 18.75rem;
+      text-align: center;
+    }
+  }
+  .nsfw-action {
+    border: none;
+    @include set-position('bottom', 'center');
```

Comment:
never saw this used before, 
can't you use a simple position and avoid the import of a mixin?